packages/mechanic-ui-components/src/input/NumberInput.module.css

```diff
@@ -24,53 +10,72 @@
 }
 
 .range-wrapper {
+  position: relative;
   display: flex;
-
   margin: 0;
   box-sizing: border-box;
-  padding: 0.5em 0;
-
+  align-items: center;
+  height: var(--mechanic-param-height);
+  padding: 0 0.5em;
   background-color: var(--mechanic-background);
+  border: 2px solid transparent;
+  color: currentColor;
+  border-radius: var(--mechanic-param-height-half);
+  & .background {
+    top: -2px;
+  }
 }
 
-.range-label {
-  flex: 0 1 90px;
-
+.range-number-input {
+  flex: 0 0 60px;
+  z-index: 10;
   box-sizing: border-box;
-  padding: 0 1em;
-
   overflow-x: hidden;
   text-overflow: ellipsis;
   white-space: nowrap;
+
+  border: none;
+  background: none;
+  font-family: inherit;
+  color: inherit;
+  font-size: inherit;
+  outline: none;
+
+  -moz-appearance: none;
+  -webkit-appearance: none;
+  appearance: none;
+
+  -moz-appearance: textfield;
+  &::-webkit-outer-spin-button,
+  &::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+    margin: 0;
+  }
 }
 
 /* https://css-tricks.com/styling-cross-browser-compatible-range-inputs-css/ */
 .range-input {
+  position: relative;
+  z-index: 9;
   flex: 1 1 auto;
   min-width: 0;
-
+  color: inherit;
   background: transparent; /* Otherwise white in Chrome */
 
```

Comment:
I dug dipper into this mysterious bug that made some test have a bigger sidebar in Firefox. I realized it was the range number input specifically, design functions that had a range number param had a much bigger sidebar only on Firefox:

![Screen Shot 2021-08-23 at 10 50 20](https://user-images.githubusercontent.com/13511560/130468349-324a818d-f44f-4e0f-b7f7-5e96577ae51c.png)
![Screen Shot 2021-08-23 at 10 50 25](https://user-images.githubusercontent.com/13511560/130468364-c7ba67b7-481b-4d64-a2b7-5ff336bed92d.png)

I did some googling and found [here](https://brennaobrien.com/blog/2014/05/style-input-type-range-in-every-browser.html) that for Firefox: "You'll also need to give input[type=range] the same width as your track. It won't expand to contain the track on it's own." I locally added here `width: 100%;` and it stopped doing the weird resizing. I'll let you decide if that makes sense here. Hope this helps!